### PR TITLE
Updated socks package to a version that doesn't use the IP package

### PR DESCRIPTION
**Pull request by shrshindeMSFT**
For more information about how to contribute to this repo, visit [this page](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md).

## Description

This repo has a CG violation for the `ip` project. The `ip` project has a published security vulnerability:
https://github.com/indutny/node-ip/issues/136 
https://github.com/advisories/GHSA-78xj-cgh5-2h22 

Unfortunately, there's no fixed version of the `ip` project. Moreover, there are questions in the "Issues" section of that project asking whether there's even a maintainer for it.

We indirectly use `ip` because we use lerna, which has a whole chain of dependencies which eventually leads to `ip`
The package in that chain which depends on `ip` is `socks.` The socks package [HAS been updated](https://github.com/JoshGlazebrook/socks/pull/94)  to stop depending on "ip" given the uncertainty of getting a fix in `ip.`

This change forces us to use the latest version of socks that removes the `ip` dependency. Eventually, over probably a while, the whole `lerna` dependency change will update to use this newer version of socks, but who knows how long that will take. In the meantime, this override ensures that with our current set of dependencies we no longer consume the vulnerable version of `ip.`

I also added a section to our package.json file that provides room to explain what each override is for, as a way of hopefully helping our future selves to know when we might be able to remove any of them. As the explanation the file says, it's a hacky way of putting comments in the file since comments are not allowed in JSON.

> If this Pull Request should close/resolve any issues when merged, use [the special syntax for that](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) here.

### Main changes in the PR:

Override `socks` package in `package.json` file.

### Unit Tests added:

> Unit tests are required for all changes. If no unit tests were added as part of this change, please explain why they aren't necessary.

N/A

### End-to-end tests added:

N/A

### Change file added:

> Ensure the change file meets the [formatting requirements](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md#change-log-using-beachball).

N/A

### CODE REVIEW

**Review by jadahiya-MSFT — APPROVED** [review]

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -1,23 +1,7 @@
 lockfileVersion: '6.0'
 
 overrides:
-  axios: ^1.6.0
-  dns-packet: ^1.3.2
-  follow-redirects: ^1.15.4
-  glob-parent: ^5.1.2
```
Comment (jadahiya-MSFT) [inline pnpm-lock.yaml:7]: Removing some of these will reintroduce old dependabot/CG issues

Comment (jadahiya-MSFT) [reply pnpm-lock.yaml:7]: I believe string-decoder and semver are the key culprits, but I'll double check and get back to you

Comment (jadahiya-MSFT) [reply pnpm-lock.yaml:7]: minimist version can't be below 0.2.4

Comment (shrshindeMSFT) [reply pnpm-lock.yaml:7]: Updated `minimist` to use latest version.

File: package.json [hunk 2]
```diff
@@ -121,5 +121,13 @@
   },
   "dependencies": {
     "uuid": "^9.0.0"
+  },
+  "pnpm": {
+    "overrides": {
+      "minimist": "^1.2.8",
+      "semver": "^7.5.4",
+      "socks": "^2.7.3",
+      "string_decoder": "^1.3.0"
```
Comment (AE-MS) [inline package.json:130]: It looks like you are adding these here but I see from the change in the lock file that some were already being overridden there -- do you know what causes that? How did there get to be overrides in the lock file that weren't listed in package.json?

File: package.json [hunk 3]
```diff
@@ -121,5 +121,13 @@
   },
   "dependencies": {
     "uuid": "^9.0.0"
+  },
+  "pnpm": {
+    "overrides": {
```
Comment (AE-MS) [inline package.json:126]: >overrides [](http://example.com/codeflow?start=5&length=9)

Is it worth considering doing something to document the justification for these overrides so people in the future know why they are here and/or when they can remove them?

**Review by AE-MS — CHANGES_REQUESTED** [review]
A few questions

**Review by jadahiya-MSFT — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (shrshindeMSFT) [reply package.json]: When migrating from `yarn` to `pnpm` the resolutions field is useful to override packages as `pnpm import` commands generates `pnpm-lock.yaml` file. Now when I added `"pnpm": {"overrides": }` field in `package.json` then `pnpm.overrides` takes precedence over `resolutions` field as mentioned in the [pnpm resolutions](https://pnpm.io/next/package_json#resolutions) documentation. To resolve this, I've updated `resolutions` to `pnpm.overrides`.

File: pnpm-lock.yaml (hunk 1 above)
Comment (shrshindeMSFT) [reply pnpm-lock.yaml]: This shouldn't be an issue anymore as `resolutions` field is updated to `pnpm.overrides`

**Review by AE-MS — APPROVED** [review]

**Review by jadahiya-MSFT — APPROVED** [review]

**Pull request merged**